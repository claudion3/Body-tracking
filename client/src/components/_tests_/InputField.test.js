import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';
import InputField from '../InputField';

import renderer from 'react-test-renderer';

afterEach(() => {
	cleanup();
});

describe('InputField component', () => {
	it('render without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<InputField />, div);
	});
	it('Render InputField', () => {
		const { getByTestId } = render(<InputField />);
		const input = getByTestId('input');
		expect(input).toBeTruthy();
	});
	it('InputField contain label', () => {
		const { queryByTestId } = render(<InputField />);
		const label = queryByTestId('label');
		expect(label).toBeFalsy();
	});
	it('InputField contain P to display error', () => {
		const { queryByTestId } = render(<InputField />);
		const error = queryByTestId('error');
		expect(error).toBeFalsy();
	});
	it('Matches snapshot', () => {
		const tree = renderer.create(<InputField />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
