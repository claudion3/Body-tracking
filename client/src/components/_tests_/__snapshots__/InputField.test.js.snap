// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component Matches snapshot 1`] = `
<div
  className="input-wrapper"
>
  <input
    autoComplete="off"
    data-testid="input"
    name=""
    onChange={[Function]}
    placeholder=""
    type="number"
    value=""
  />
  
</div>
`;
